redbot/core/cog_manager.py

```diff
@@ -209,10 +211,10 @@ def _ensure_path_obj(path: Union[Path, str]) -> Path:
 
         Raises
         ------
-        RuntimeError
-            When no matching spec can be found.
+        NoSuchCog
+            When no cog with the requested name was found.
         """
-        resolved_paths = _deduplicate(await self.paths())
+        resolved_paths = deduplicate_iterables(await self.paths())
```

Comment:
`paths()` already deduplicates the paths before it returns, so you can remove the call to deduplicate here